lib/web_ui/lib/src/engine/window.dart

```diff
@@ -29,8 +31,17 @@ const bool debugPrintPlatformMessages = false;
 /// The view ID for the implicit flutter view provided by the platform.
 const int kImplicitViewId = 0;
 
+/// Represents all views in the Flutter Web Engine.
```

Comment:
I think that the class that represents all views in the Flutter Web Engine is `EngineFlutterWindow` (that's the type that was used to define the `implicitView`, for example), right?

Reply:
`EngineFlutterWindow` is a misnomer. It should be called `EngineSingletonFlutterWindow` (it actually extends `ui.SingletonFlutterWindow`). Views in the multi-view world shouldn't inherit from this singleton class. They should inherit directly from `ui.FlutterView` (and implement `EngineFlutterView` to satisfy all the web-specific needs).

Reply:
Here's a follow up PR to rename this class to make it clear that it's a singleton: https://github.com/flutter/engine/pull/45981